docs/core/build/intro-the-stack.md

```diff
@@ -0,0 +1,55 @@
+
+As with any virtual machine, the VM creates a level of abstraction between the executing code and the executing machine (an Conflux node). Currently, the VM is running on thousands of nodes distributed across the world.
+
+Under the hood, the VM uses a set of opcode instructions to execute specific tasks. These (140 unique) opcodes allow the VM to be [Turing-complete](https://en.wikipedia.org/wiki/Turing_completeness), which means the VM is able to compute just about anything, given enough resources.
+
+As a dapp developer, you don't need to know much about the VM other than it exists and that it reliably powers all applications on Conflux without downtime.
+
+## Level 2: Smart contracts
+
+[Smart contracts](./smart-contracts) are the executable programs that run on the Conflux Core blockchain.
+
+Smart contracts are written using specific programming languages(Solidity) that compile to EVM bytecode (low-level machine instructions called opcodes).
+
+Not only do smart contracts serve as open source libraries, they are essentially open API services that are always running and can't be taken down. Smart contracts provide public functions which users and applications (dapps) may interact with, without needing permission. Any application may integrate with deployed smart contracts to compose functionality, such as adding data feeds or to support token swaps. Additionally, anyone can deploy new smart contracts to Conflux in order to add custom functionality to meet their application's needs.
+
+As a dapp developer, you'll need to write smart contracts only if you want to add custom functionality on the Conflux Core blockchain. You may find you can achieve most or all of your project's needs by merely integrating with existing smart contracts, for instance if you want to support payments in stablecoins or enable decentralized exchange of tokens.
+
+## Level 3: Conflux nodes
+
+In order for an application to interact with the Conflux Core blockchain, it must connect to an [Conflux node](../../general/run-a-node/Overview.md). Connecting to a node allows you to read blockchain data and/or send transactions to the network.
+
+Conflux nodes are computers running software - an Conflux client. A client is an implementation of Conflux that verifies all transactions in each block, keeping the network secure and the data accurate. **Conflux nodes are the Conflux blockchain**. They collectively store the state of the Conflux blockchain and reach consensus on transactions to mutate the blockchain state.
+
+By connecting your application to an Conflux node (via the [JSON-RPC API](./json-rpc/)), your application is able to read data from the blockchain (such as user account balances) as well as broadcast new transactions to the network (such as transferring CFX between user accounts or executing functions of smart contracts).
+
+## Level 4: Conflux Core Client APIs
```

Comment:
I'm not sure if it's ok to list SDKs as a level. It is not necessary for apps to use sdks to connect to nodes and sdks do not only perform to connect to nodes but also provides helpful utils to simplify development. 

From my point of view, a better way is to mention both in nodes part and applications part that developers could use SDKs to finish certain tasks.